scipy/sparse/__init__.py

```diff
@@ -32,6 +32,8 @@
      that all *slicing* operations on these objects must produce
      two-dimensional results, or they will result in an error. This
      will be addressed in a future version.
+     Currently *slicing* is not supported by `bsr_array`, `coo_array`,
+     `dia_array`.
```

Comment:
I don't believe this note belongs in this place in the docs. It is not a difference between array and matrix interfaces. And we don't include a description of all the many differences in support between the formats here. 

So, I would prefer we remove this note, and focus on the doc_strings for the individual formats.